File: app/src/main/java/com/github/liketechnik/mentalMathTrainer/CheckYourSettings.java
package com.github.liketechnik.mentalMathTrainer;

import android.app.Dialog;
import android.content.DialogInterface;
import android.content.Intent;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.Switch;
import com.github.liketechnik.mentalmathtrainer.R;

public class CheckYourSettings extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_check_your_settings);

        Intent intent = getIntent();

        boolean addition = intent.getBooleanExtra(NewTrainingSession.ADDITION, true);
        int maxAdditionValue = intent.getIntExtra(NewTrainingSession.MAX_ADDITION_VALUE,
                NewTrainingSession.defaultMaxAdditionValue);
        int numberOfExercises = intent.getIntExtra(NewTrainingSession.NUMBER_OF_EXERCISES,
                NewTrainingSession.defaultNumberOfExercises);

        Switch additionSwitch = (Switch) findViewById(R.id.addition_switch_check);
        additionSwitch.setChecked(addition);

        EditText maxAdditionValueEditText = (EditText) findViewById(R.id.max_addition_value_check);
        maxAdditionValueEditText.setText(String.valueOf(maxAdditionValue));

        EditText numberOfExercisesEditText = (EditText) findViewById(R.id.number_of_exercises_check);
        numberOfExercisesEditText.setText(String.valueOf(numberOfExercises));


    }

    /** Called when the user edits the Maximum addition value */
    public void onClickMaxAdditionValue (View view) {
        Intent intent = getIntent();

        int maxAdditionValue = intent.getIntExtra(NewTrainingSession.MAX_ADDITION_VALUE,
                NewTrainingSession.defaultMaxAdditionValue);

        EditText maxAdditionValueEditText = (EditText) findViewById(R.id.max_addition_value_check);
        maxAdditionValueEditText.setText(String.valueOf(maxAdditionValue));

        DialogFragment helpDialog = new noEditingHelpDialogFragment();
        helpDialog.show(getSupportFragmentManager(), "noEditingHelpDialogFragment");
    }

    /** Called when the user edits the Number fo Exercises */
    public void onClickNumberOfExercises (View view) {
        Intent intent = getIntent();

        int numberOfExercises = intent.getIntExtra(NewTrainingSession.NUMBER_OF_EXERCISES,
                NewTrainingSession.defaultNumberOfExercises);

        EditText numberOfExercisesEditText = (EditText) findViewById(R.id.number_of_exercises_check);
        numberOfExercisesEditText.setText(String.valueOf(numberOfExercises));

        DialogFragment helpDialog = new noEditingHelpDialogFragment();
        helpDialog.show(getSupportFragmentManager(), "noEditingHelpDialogFragment");
    }

    /** Called when the user clicks on the Addition switch */
    public void onClickAdditionSwitch(View view) {
        Intent intent = getIntent();

        boolean addition = intent.getBooleanExtra(NewTrainingSession.ADDITION, true);

        Switch additionSwitch = (Switch) findViewById(R.id.addition_switch_check);
        additionSwitch.setChecked(addition);

        DialogFragment helpDialog = new noEditingHelpDialogFragment();
        helpDialog.show(getSupportFragmentManager(), "noEditingHelpDialogFragment");
    }

    /** Builder for the help dialog about not editing */
    static public class noEditingHelpDialogFragment extends DialogFragment {
        @Override
        public Dialog onCreateDialog(Bundle savedInstanceState) {
            AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
            builder.setMessage(R.string.no_editing_help)
                    .setTitle(R.string.no_editing_help_title)
                    .setPositiveButton(R.string.ok, new DialogInterface.OnClickListener() {
                        public void onClick (DialogInterface dialog, int id) {
                            // Do nothing, user knows what he wanted to know
                        }
                    });

            return builder.create();
        }
    }
}

File: app/src/main/java/com/github/liketechnik/mentalMathTrainer/CheckYourSettings.kt
package com.github.liketechnik.mentalMathTrainer

import android.app.Dialog
import android.content.DialogInterface
import android.content.Intent
import android.support.v4.app.DialogFragment
import android.support.v7.app.AlertDialog
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.widget.Switch
import com.github.liketechnik.mentalMathTrainer.R
import com.github.liketechnik.mentalMathTrainer.utils.TrainingSessionConstants.Companion.ADDITION
import com.github.liketechnik.mentalMathTrainer.utils.TrainingSessionConstants.Companion.MAX_ADDITION_VALUE
import com.github.liketechnik.mentalMathTrainer.utils.TrainingSessionConstants.Companion.NUMBER_OF_EXERCISES
import com.github.liketechnik.mentalMathTrainer.utils.TrainingSessionConstants.Companion.defaultMaxAdditionValue
import com.github.liketechnik.mentalMathTrainer.utils.TrainingSessionConstants.Companion.defaultNumberOfExercises
import com.github.liketechnik.mentalMathTrainer.utils.applyTrainingSessionIntent
import com.github.liketechnik.mentalMathTrainer.utils.packTrainingSessionIntent


class CheckYourSettings : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_check_your_settings)

        applyTrainingSessionIntent(this)
    }

    /** Called when the user edits the Maximum addition value  */
    fun onClickMaxAdditionValue(view: View) {
        val intent = intent

        val maxAdditionValue = intent.getIntExtra(MAX_ADDITION_VALUE,
                defaultMaxAdditionValue)

        val maxAdditionValueEditText = findViewById(R.id.max_addition_value) as EditText
        maxAdditionValueEditText.setText(maxAdditionValue.toString())

        val helpDialog = noEditingHelpDialogFragment()
        helpDialog.show(supportFragmentManager, "noEditingHelpDialogFragment")
    }

    /** Called when the user edits the Number fo Exercises  */
    fun onClickNumberOfExercises(view: View) {
        val intent = intent

        val numberOfExercises = intent.getIntExtra(NUMBER_OF_EXERCISES,
                defaultNumberOfExercises)

        val numberOfExercisesEditText = findViewById(R.id.number_of_exercises) as EditText
        numberOfExercisesEditText.setText(numberOfExercises.toString())

        val helpDialog = noEditingHelpDialogFragment()
        helpDialog.show(supportFragmentManager, "noEditingHelpDialogFragment")
    }

    /** Called when the user clicks on the Addition switch  */
    fun onClickAdditionSwitch(view: View) {
        val intent = intent

        val addition = intent.getBooleanExtra(ADDITION, true)

        val additionSwitch = findViewById(R.id.addition_switch) as Switch
        additionSwitch.isChecked = addition

        val helpDialog = noEditingHelpDialogFragment()
        helpDialog.show(supportFragmentManager, "noEditingHelpDialogFragment")
    }

    /** Builder for the help dialog about not editing  */
   inner class noEditingHelpDialogFragment : DialogFragment() {
        override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
            val builder = AlertDialog.Builder(activity)
            builder.setMessage(R.string.no_editing_help)
                    .setTitle(R.string.no_editing_help_title)
                    .setPositiveButton(R.string.ok) { _, _ ->
                        // Do nothing, user knows what he wanted to know
                    }
                    .setNegativeButton(R.string.back) { _, _ ->
                        startActivity(packTrainingSessionIntent(this@CheckYourSettings, NewTrainingSession::class.java))
                    }
            return builder.create()
        }
    }
}
